registry/storage/driver/testsuites/testsuites.go

```diff
@@ -1224,3 +1239,27 @@ func firstPart(filePath string) string {
 		filePath = dir
 	}
 }
+
+type errReader struct {
+	n   int64
+	msg string
+}
+
+func (er *errReader) Read(p []byte) (int, error) {
+	toread := int64(len(p))
```

Comment:
What is the exact purpose of this argument?  Could you just return `0, fmt.Errorf(er.msg)` without all the writing..?